optaplanner-core/src/main/java/org/optaplanner/core/api/score/constraint/ConstraintMatch.java

```diff
@@ -81,8 +82,11 @@ public int compareTo(ConstraintMatch other) {
         } else if (!constraintName.equals(other.constraintName)) {
             return constraintName.compareTo(other.constraintName);
         } else {
+            Comparator<Object> comparator = new ConstraintJustificationComparator();
```

Comment:
That ConstraintJustificationComparator caches the reflection results, but for every ConstriantMatch (= 1000's) we create a new one?
I am ok with merging this as is right now, but we need a new jira to improve performance of constraint match enabled by somehow having this comparator once on the scoreHolder and scoreInliner? Idunno

Reply:
@triceo Can you add a TODO comment to point out the problem?